core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorSplitManager.java

```diff
@@ -30,7 +30,8 @@ default ConnectorSplitSource getSplits(
             ConnectorSession session,
             ConnectorTableHandle table,
             SplitSchedulingStrategy splitSchedulingStrategy,
-            DynamicFilter dynamicFilter)
+            DynamicFilter dynamicFilter,
+            ConstraintEvaluator constraintEvaluator)
```

Comment:
This is a breaking SPI change. It should be handled by leaving the old method as `@Deprecated` and introducing the new method, with default implementation calling the old one.

This would also let you avoid all the changes across connectors in this PR. Of course, they should be migrated still, but doing them in a follow-up helps validate that SPI backward compatibility is maintained.

Reply:
ok I considered that but after looking at implementation of dynamic filtering I thought it was done the same way I did.